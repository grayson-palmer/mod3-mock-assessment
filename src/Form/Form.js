import React, { Component } from 'react';
import './Form.scss';

export default class Form extends Component {
  constructor() {
    super();
    this.state = {};
  }

  handleChange(e) {
    this.setState({ [e.target.name]: e.target.value })
  }

  resetInputs() {
    this.setState({
      name: '',
      date: '',
      time: '',
      number: ''
    })
  }

  render() {
    return (<div className='form'>
      <input 
        name='name'
        type='text'  
        placeholder='Name' 
        className='form__name'
        onChange={ e => {this.handleChange(e)}}
      />
      <input 
        name='date'
        type='text' 
        placeholder='Date (mm/dd)' 
        className='form__name'
        onChange={ e => {this.handleChange(e)}}
      />
      <input 
        name='time'
        type='text' 
        placeholder='Time' 
        className='form__name'
        onChange={ e => {this.handleChange(e)}}
      />
      <input 
        name='number'
        type='text' 
        placeholder='Number of Guests' 
        className='form__name'
        onChange={ e => {this.handleChange(e)}}
      />
      <button 
        type='button' 
        className='form__button'
        onClick={() => {
          this.props.addReservation(this.state)
          this.resetInputs()}}>
            Make Reservation
        </button>
    </div>
    )
  }
}