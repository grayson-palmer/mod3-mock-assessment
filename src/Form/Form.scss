.form {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-around;
  margin: 2em;
  width: 100%;
  > input {
    font-size: 1em;
    padding: 0.25em;
    width: 12em;
  }
  &__button {
    background-color: teal;
    color: white;
    font-size: 1em;
    width: 12em;
  }
}