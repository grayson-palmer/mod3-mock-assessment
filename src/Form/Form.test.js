import React from 'react';
import Form from './Form.js';
import { shallow } from 'enzyme';

describe('Form', () => {
  let wrapper;
  const mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form addReservation={mockAddReservation} />);
  })

  it ('should be a snapshot of the form element', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it ('should handle changes in the state', () => {
    const startState = {
      name: 'Grayson',
      date: '03/08',
      time: '8:00',
      number: 2
    }
    wrapper.setState(startState);
    wrapper.instance().resetInputs();
    expect(wrapper.state('title')).toEqual('');
  })
})