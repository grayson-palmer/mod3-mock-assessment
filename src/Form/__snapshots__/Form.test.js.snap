// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should be a snapshot of the form element 1`] = `
<div
  className="form"
>
  <input
    className="form__name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
  />
  <input
    className="form__name"
    name="date"
    onChange={[Function]}
    placeholder="Date (mm/dd)"
    type="text"
  />
  <input
    className="form__name"
    name="time"
    onChange={[Function]}
    placeholder="Time"
    type="text"
  />
  <input
    className="form__name"
    name="number"
    onChange={[Function]}
    placeholder="Number of Guests"
    type="text"
  />
  <button
    className="form__button"
    onClick={[Function]}
    type="button"
  >
    Make Reservation
  </button>
</div>
`;
